db/model/generator.js

```diff
@@ -332,6 +332,45 @@ module.exports = function generator(seq, dataTypes) {
             resolve(found.length === 1);
           }));
       }, // isWritableBy
+
+      /**
+       * Replaces array of subject absolutePaths with array of subject records.
+       * Replaces array of aspect names with array of aspect records.
+       * TODO: set array of subjects when generator provides "subjectQuery"
+       *
+       * @returns {Generator} - the updated Generator
+       */
+      updateForHeartbeat() {
+        const g = this.get();
+        const aspectPromises = g.aspects.map((a) =>
+          seq.models.Aspect.findOne({ where: { name: { $iLike: a } } }));
```

Comment:
We have to find only published aspects , correct?

Reply:
Correct... and I'll make that change here... but we should never actually get into that state where the Generator references an unpublished aspect because we will be preventing unpublish/delete of aspects which are referenced by a SG.